tools/bazar/services/BazarListService.php

```diff
@@ -0,0 +1,244 @@
+<?php
+
+namespace YesWiki\Bazar\Service;
+
+use YesWiki\Bazar\Controller\EntryController;
+use YesWiki\Wiki;
+use YesWiki\Bazar\Field\BazarField;
+use YesWiki\Bazar\Service\EntryManager;
+use YesWiki\Bazar\Service\ExternalBazarService;
+use YesWiki\Bazar\Service\FormManager;
+
+class BazarListService
+{  
+    public function __construct(Wiki $wiki, EntryManager $entryManager, EntryController $entryController,
+                                ExternalBazarService $externalBazarService, FormManager $formManager)
+    {
+      $this->wiki = $wiki;
+      $this->entryManager = $entryManager;
+      $this->entryController = $entryController;
+      $this->externalBazarService = $externalBazarService;
+      $this->formManager = $formManager;
+    }
+
+    public function getForms($options) : array
+    {
+        // External mode activated ?
+        if ($options['externalModeActivated'] === true) {
+            return $this->externalBazarService
+                        ->getFormsForBazarListe($options['externalIds'], $options['refresh']);
+        } else {
+            return $this->formManager->getAll();
+        }
+    }
+
+    public function getEntries($options, $forms = null) : array
+    {
+        if (!$forms) $forms = $this->getForms($options);
+
+        // External mode activated ?
+        // TODO BazarListdynamic test externalmode works
+        if ($options['externalModeActivated'] === true) {
+            $entries = $this->externalBazarService->getEntries([
+                'forms' => $forms,
+                'refresh' => $options['refresh'],
+                'queries' => $options['query']
+            ]);
+        } else {
+            $entries = $this->entryManager->search(
+                [
+                    'queries' => $options['query'] ?? '',
+                    'formsIds' => $options['idtypeannonce'],
+                    'keywords' => $_REQUEST['q'] ?? '',
+                    'user' => $options['user'],
+                    'minDate' => $options['dateMin'],
+                    'correspondance' => $options['correspondance'] ?? ''
+                ],
+                true // filter on read ACL
+            );
+            
+        }
+
+        // filter entries on datefilter parameter
+        if (!empty($options['datefilter'])) {
+            $entries = $this->entryController->filterEntriesOnDate($entries, $options['datefilter']) ;
+        }
+
+        // Sort entries
+        if ($options['random']) {
+            shuffle($entries);
+        } else {
+            usort($entries, $this->buildFieldSorter($options['ordre'], $options['champ']));
+        }
+
+        // Limit entries
+        if ($options['nb'] !== '') {
+            $entries = array_slice($entries, 0, $options['nb']);
+        }
+
+        return $entries;
+    }
+
+    public function formatFilters($options, $entries, $forms) : array
```

Comment:
```suggestion
    private function formatFilters($options, $entries, $forms) : array
```

je ne suis pas partisan pour rendre cette méthode publique et je n'en vois pas l'intérêt.
Comme l'appel de la méthode getEntries et toujours conjointe à cette méthode, je ne ferai u'une seule méthode publique qui rend le résultats des deux méthodes.

idem je préfère garder l'appellation `$arguments`